protocol/rpcprovider/provider_load_manager.go

```diff
@@ -0,0 +1,59 @@
+package rpcprovider
+
+import (
+	"context"
+	"strconv"
+	"sync/atomic"
+
+	"github.com/lavanet/lava/v3/protocol/chainlib"
+	grpc "google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+)
+
+type ProviderLoadManager struct {
+	rateLimitThreshold      atomic.Uint64
+	activeRequestsPerSecond atomic.Uint64
+}
+
+func NewProviderLoadManager(rateLimitThreshold uint64) *ProviderLoadManager {
+	if rateLimitThreshold == 0 {
+		return nil
+	}
+	loadManager := &ProviderLoadManager{}
+
+	loadManager.rateLimitThreshold.Store(rateLimitThreshold)
+
+	return loadManager
+}
+
+func (loadManager *ProviderLoadManager) addRelayCall() {
+	if loadManager == nil {
+		return
+	}
+	loadManager.activeRequestsPerSecond.Add(1)
+}
+
+func (loadManager *ProviderLoadManager) subtractRelayCall() {
+	if loadManager == nil {
+		return
+	}
+	loadManager.activeRequestsPerSecond.Add(^uint64(0))
```

Comment:
Why the `^`?

Reply:
Fixed in [44a5e5c](https://github.com/lavanet/lava/pull/1720/commits/44a5e5c1624b10f6a2bdc7dffd3b354a8203b0be)